doc/source/how-to/deprecating.rst

```diff
@@ -0,0 +1,126 @@
+This process helps maintain clarity and consistency across the projects.
+
+Maintainer tasks
+----------------
+
+If you are a maintainer of a library that is being deprecated, please follow these steps. As
+an illustrative example, the deprecation of `PyAdditive Widgets <https://github.com/ansys/pyadditive-widgets>`_
+is used in this guide.
+
+1. **Inform the PyAnsys Core team**: Before proceeding with the deprecation, it is essential to
+   inform the PyAnsys Core team. This can be done by sending an email to
+   the team at `pyansys.core@ansys.com <pyansys_core_email_>`_. This step ensures that the
+   deprecation is communicated effectively and that the team can assist with any necessary
+   changes in the PyAnsys ecosystem.
+
+2. **Close all open issues and pull requests**: Before deprecating the library, ensure that all
+   open issues and pull requests in the library's repository are closed. This helps to avoid
+   confusion and ensures that users are aware that the library is no longer actively maintained.
+   You can close issues and pull requests with a comment explaining that the library is being
+   deprecated and is no longer be supported.
+
+3. **Create a deprecation issue**: Open an issue in the library's repository to announce the
+   deprecation. This issue should explain why the library is being deprecated, what alternatives
+   are available, and any relevant timelines. See a template below for the issue content.
+
+   .. warning::
+
+      Make sure to adapt the template to your specific library and situation.
+
+   .. code:: markdown
+
+      ## ⚠️ Project Deprecation Notice
+
+      **This repository is no longer maintained as of [DATE].**
+
+      ### Reason for deprecation
+      [Explain briefly why the project is being deprecated — e.g. better alternatives, no time to maintain, outdated use case.]
+
+      ### Alternatives
+      If you're looking for a maintained alternative, consider:
+      - [Alternative 1](https://...)
+      - [Alternative 2](https://...)
+
+      ### What this means
+      - No further updates, bug fixes, or pull request reviews
+      - Issues will be closed
+      - The repository will be archived
+
+      Thank you to everyone who contributed, used, or supported this project!
+
+   This issue serves as a permanent record of the deprecation and provide users with
+   necessary information about alternatives.
+
+   Make sure to pin the issue to the top of the repository so that it is easily visible to users.
+   This can be done by selecting the "Pin issue" option in the issue's right-side menu, on GitHub.
+
+   See an example at `PyAdditive Widgets deprecation issue`_.
+
+4. **Adapt the README**: Update the library's ``README`` file to reflect the deprecation.
+   This should include a clear notice at the top of the ``README``, informing users that the
+   library is deprecated. See an example at `PyAdditive Widgets README`_.
```

Comment:
Should we also make a modification in the documentation landing page?

Reply:
That's a good point - I like it. I think we should!